src/lib.rs

```diff
@@ -0,0 +1,270 @@
+use log::trace;
+use std::{collections, fs, path};
+
+pub struct OverdropConf {
+    dirs: Vec<path::PathBuf>,
+}
+
+impl OverdropConf {
+    pub fn new(
+        base_dirs: &Vec<String>,
+        config_path: &str,
+    ) -> Self {
+        let mut dirs = Vec::with_capacity(base_dirs.len());
+        for bdir in base_dirs {
+            let mut dpath = path::PathBuf::from(bdir);
+            dpath.push(config_path.clone());
+            dirs.push(dpath);
+        }
+        Self { dirs }
+    }
+
+    pub fn new_full(
+        root_dir: &str,
+        base_dirs: &Vec<String>,
+        config_path: &str,
+        version: Option<String>,
+    ) -> Self {
+        let mut dirs = Vec::with_capacity(base_dirs.len());
+        let version = version.unwrap_or("".to_string());
+        for bdir in base_dirs {
+            let mut dpath = path::PathBuf::from(root_dir);
+            dpath.push(bdir);
+            dpath.push(format!("{}{}", config_path, version));
+            dirs.push(dpath);
+        }
+        Self { dirs }
+    }
+
+    pub fn scan_unique_files(
+        &self,
+        allow_hidden: bool,
+        allowed_extensions: Option<&Vec<&str>>,
+    ) -> failure::Fallible<collections::BTreeMap<String, path::PathBuf>> {
+        let mut files_map = collections::BTreeMap::new();
+        for dir in &self.dirs {
+            trace!("Scanning directory '{}'", dir.to_str().unwrap());
+
+            let dir_iter = match fs::read_dir(dir) {
+                Ok(iter) => iter,
+                _ => continue,
+            };
+            for dir_entry in dir_iter {
+                let entry = match dir_entry {
+                    Ok(f) => f,
+                    _ => continue,
+                };
+                let fpath = entry.path();
+                let fname = entry.file_name().into_string().unwrap();
+
+                // If hidden files not allowed, ignore dotfiles.
+                if !allow_hidden && fname.starts_with('.') {
+                    continue;
+                };
+
+                // If extensions are specified, proceed only if filename
+                // has one of the allowed extensions.
+                if let Some(allowed) = allowed_extensions {
+                    if let Some(e) = fpath.extension() {
+                        if !allowed.contains(&e.to_str().unwrap()) {
```

Comment:
Let's `continue` instead of unwrapping here.

Reply:
Fixed